adapters/repos/db/index.go

```diff
@@ -751,12 +760,30 @@ func (i *Index) putObjectBatch(ctx context.Context, objects []*storobj.Object,
 	}
 
 	byShard := map[string]objsAndPos{}
+	// get all tenants shards
+	tenants := []string{}
```

Comment:
change(performance): We can preallocate the `tenants` array here with `len(objects)`. Worst case we over allocate if the tenant are empty I think this is acceptable.